bazel/linux/defs.bzl

```diff
@@ -312,29 +324,307 @@ Example:
+        name = "test-latest-rootfs",
+        package = "buildroot-custom-amd64",
+        url = "astore.corp.enfabrica.net/d/kernel/test/buildroot-custom-amd64.img",
+    )
+
+To download the specified image from "https://astore.corp.enfabrica.net/d/kernel",
+and use it as the "test-latest-rootfs" from the repository.
+""",
+    implementation = _rootfs_version,
+    local = False,
+    attrs = {
+        "package": attr.string(
+            doc = "The name of the downloaded image. Usually the format is 'distribution-rootfs_version-arch', like buildroot-custom-amd64.",
+            mandatory = True,
+        ),
+        "url": attr.string(
+            doc = "The url to download the rootfs image from.",
+            mandatory = True,
+        ),
+        "sha256": attr.string(
+            doc = "The sha256 of the downloaded package file.",
+        ),
+        "auth": attr.string_dict(
+            doc = "An auth dict as documented for the download_and_extract context rule as is.",
+        ),
+        "_template": attr.label(
+            default = Label("//bazel/linux:rootfs.BUILD.bzl"),
+            allow_single_file = True,
+        ),
+        "_utils": attr.label(
+            default = Label("//bazel/linux:defs.bzl"),
+            allow_single_file = True,
+        ),
+    },
+)
+
+KernelImageInfo = provider(
+    doc = """Maintains the information necessary to represent a kernel executable image.""",
+    fields = {
+        "name": "Name of the rule that defined this kernel executable image. For example, 'stefano-s-favourite-kernel-image'.",
+        "package": "A string indicating which package this kernel executable image is coming from. For example, 'custom-5.9.0-um'.",
+        "image": "Path of the kernel executable image.",
+    },
+)
+
+def _kernel_image(ctx):
+    return [DefaultInfo(files = depset([ctx.file.image])), KernelImageInfo(
+        name = ctx.attr.name,
+        package = ctx.attr.package,
+        image = ctx.file.image,
+    )]
+
+kernel_image = rule(
+    doc = """Defines a new kernel executable image.
+
+This rule exports a file that represent a kernel executable image with just
+enough to be able to run kernel tests.
+
+kernel_image rules are typically automatically created when you declare a
+kernel_image_version() in your WORKSPACE file. You should almost never have to
+create kernel_image rules manually.
+
+The only exception is if you want to troubleshoot a new kernel image you have
+available locally.
+
+Example:
+
+    kernel_image(
+        # An arbitrary name for the rule.
+        name = "stefano-s-favourite-kernel-image",
+        # This kernel image file.
+        image = "custom-5.9.0-um",
+    )
+""",
+    implementation = _kernel_image,
+    attrs = {
+        "package": attr.string(
+            mandatory = True,
+            doc = "A string indicating which package this kernel executable image is coming from.",
+        ),
+        "image": attr.label(
+	    mandatory = True,
```

Comment:
nit: indentation, run `bazelisk run //:buildifier` to fix indentation and style for you.